### ISSUE TITLE
[QUESTION] Is it planned to ugrade to Gtk4 when that's possible?

### ISSUE THREAD

**Issue by rempas** [gnunn1/tilix#2013]
First of I would like to congratulate you for your awesome work and for choosing the D programming language to create your project! I would like to ask, is it planned to upgrade Tilix to Gtk4 once Gtkd also supports it?

**Comment by ximion** [comment]
That depends on whether there will be a new full-time maintainer who can take on the work - see https://github.com/gnunn1/tilix/issues/1700

**Comment by talheim-it** [comment]
@ximion  any option to promote to get developer for tilix? Like to fund it via patreon or so?

I use tilix in my daily business on Fedora and never want to miss it. If I can help somehow to find developers and funding it with the community, that should be a possible thing or?

**Comment by Jiehong** [comment]
So far, this is not possible, since gtkd is not yet gtk 4 ready (see https://github.com/gtkd-developers/GtkD/issues/315).

However, tilix could be updated to GTKD 3.10.0 from 3.9.0 (gtkd releases: https://github.com/gtkd-developers/GtkD/tags)

**Comment by rempas** [comment]
> So far, this is not possible, since gtkd is not yet gtk 4 ready (see [gtkd-developers/GtkD#315](https://github.com/gtkd-developers/GtkD/issues/315)).
> 
> However, tilix could be updated to GTKD 3.10.0 from 3.9.0 (gtkd releases: https://github.com/gtkd-developers/GtkD/tags)

Yes of course, this was made a little bit more than 1 year ago. I hope everyone is happy and having fun from their side!

**Comment by yookoala** [comment]
Just placed a bounty on https://github.com/gtkd-developers/GtkD/issues/315, please help contribute the bounty to encourage developer(s) to implement it.

**Comment by patriziobruno** [comment]
@rempas why did you close this as complete? Neither GTKD nor Tilix were ever ported to gtk4

**Comment by rempas** [comment]
> @rempas why did you close this as complete? Neither GTKD nor Tilix were ever ported to gtk4

Idk. Should I re-open it? I guess, if the developers working on it, there is no point in having the issue open.

**Comment by patriziobruno** [comment]
well, issues are to track work in progress, they're not a bad thing :) they're supposed to be closed when the work is complete or if the request was rejected

**Comment by rempas** [comment]
> well, issues are to track work in progress, they're not a bad thing :) they're supposed to be closed when the work is complete or if the request was rejected

Oh, ok! I'll have than in mind, thank you!

**Comment by Arcitec** [comment]
GtkD is abandoned in 2021 and doesn't seem like it will support Gtk4. There's a branch for it which was last updated 4 years ago:

https://github.com/gtkd-developers/GtkD/tree/gtk4

Unfortunately that makes it extremely hard and unlikely to support Gtk4 in Tilix.

You can use this theme to make Tilix look like it was made in Gtk4 though:

https://github.com/lassekongo83/adw-gtk3

**Comment by rempas** [comment]
> GtkD is abandoned in 2021 and doesn't seem like it will support Gtk4. There's a branch for it which was last updated 4 years ago:
> 
> https://github.com/gtkd-developers/GtkD/tree/gtk4
> 
> Unfortunately that makes it extremely hard and unlikely to support Gtk4 in Tilix.
> 
> You can use this theme to make Tilix look like it was made in Gtk4 though:
> 
> https://github.com/lassekongo83/adw-gtk3

The package [GID](https://github.com/Kymorphia/gid) can also be used! Tho, tilix will have to be modified to use it.
